### PR TITLE
Add tensor.numel

### PR DESCRIPTION
**Pull request by nima10khodaveisi**

### CODE REVIEW

**Comment by nima10khodaveisi** [comment]
I can add rest of them as well, #861.

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -158,6 +158,8 @@ def arange(stop, start=0, step=1, **kwargs): return Tensor(np.arange(start=start
   def where(self:Tensor, input_:Union[Tensor, float], other:Union[Tensor, float]):
     cond = (self != 0.0)
     return cond * input_ + (1.0 - cond) * other
+  
+  def numel(self): return functools.reduce(lambda x,y:x*y,self.shape)
```
Comment (wozeparrot) [inline tinygrad/tensor.py:162]: Can this not just be `math.prod(self.shape)`?

Comment (nima10khodaveisi) [reply tinygrad/tensor.py:162]: yeah, sure. will change it.

**Comment by xl0** [comment]
Numel should be a @property to match torch and numpy.

**Comment by xl0** [comment]
Please disregard the last comment, I was thinking `.ndim` :man_facepalming:

**Pull request merged**